cardstack/src/screens/PayMerchant/usePayMerchant.ts

```diff
@@ -177,7 +177,11 @@ export const usePayMerchant = () => {
     qrCodeNetwork,
   } = data;
 
-  const { paymentChangeCurrency, currency: nativeCurrency } = usePayment();
+  const {
+    paymentChangeCurrency,
+    currency: nativeCurrency = initialCurrency || NativeCurrency.SPD,
+  } = usePayment();
+
```

Comment:
```suggestion
  const {
    paymentChangeCurrency,
    currency
  } = usePayment();
  const nativeCurrency = currency || initialCurrency || NativeCurrency.SPD;
```

Personally, I find this a lot easier to read than the destructuring assignment with new variable name plus default values. If you and @danibonilha prefer the original though, I can get used to it.